server/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -181,6 +193,34 @@ public int id() {
         return id;
     }
 
+    public String handlerNode() {
+        return handlerNode;
+    }
```

Comment:
I'm not sure about having the handler node in the session. It feels like it doesn't belong here and seems a bit redundant to place this into the session given each node knows this and could evaluate it based on the cluster state as part of the system table expression.

Would it be a hassle to move it there?
Not a dealbreaker.

Reply:
done, thx!